README.md

```diff
@@ -0,0 +1,117 @@
+        app.resetInjection()
+        app.overrideModule = testDependencies
+    }
+```
+
+First of all, we are initiating all the provisioning mocks with Mockito. After that, we use an extension method that we have in the sample Android application to obtain the application. First, we clean the graph, we like have all dependencies clean before each test, to be repeatable. At last, we save our test module to replace the regular module dependencies.
+
+We need to save the module because we don't have control over the injections. the order of modules has been injected is important. We like that first injected the application modules, after that the activity modules and in the end, the testing modules overriding the previous provisions. When we invoke the start activity, Kodein is going to create the module and injected it in the graph, and we need be ready to apply our testing replaces.
+
+If you check ```KodeinSampleApp.kt```
+
+```java
+class KodeinSampleApp : Application(), KodeinAware {
+    override val kodein = ConfigurableKodein(mutable = true)
+    var overrideModule: Module? = null
+
+    override fun onCreate() {
+        super.onCreate()
+        resetInjection()
+    }
+
+    fun addModule(activityModules: Module) {
+        kodein.addImport(activityModules, true)
+        if (overrideModule != null) {
+            kodein.addImport(overrideModule!!, true)
+        }
+    }
+
+    fun resetInjection() {
+        kodein.clear()
+        kodein.addImport(appDependencies(), true)
+    }
+
+    private fun appDependencies(): Module {
+        return Module(allowSilentOverride = true) {
+            bind<ApplicationScopeClass>() with singleton {
+                ApplicationScopeClass()
+            }
+        }
+    }
+}
+```
+
+We have an instance of Kodein that allows us to have to mutate the graph and compose in different modules.
+
+You can check how to resetInjection, clear the graph and reinject the application dependencies, for each test, We have everything clear for each invocation.
+
+The method addModule has been called for the Activities to inject their modules, this method checks if we have a Module that should be added to the graph after the activity module. This going to happen when we invoke the tests, remember that before each test we assign to override module our testing module with our mocks.
+
+The last step to check that we need to is in the ```MainActivity.kt```.
+
+```java
+    override fun onCreate(savedInstanceState: Bundle?) {
+        applicationContext.asApp().addModule(activityModules)
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        setTextFromController()
+    }
+```
+
+We are accessing the Application for injecting our activity module before any execution in the activity.
+
+With those simple steps, We can override our dependencies in those tests where we don't have access to the Constructor, like activities or services.
```

Comment:
With these simple steps, we can override our dependencies in our tests! We don't have access to the Activities or Services constructors in order to replace production-code dependencies with test doubles.

Reply:
checked